testplan/web_ui/testing/src/Report/BatchReportBeta/state/reportSlice.js

```diff
@@ -0,0 +1,65 @@
+// @ts-nocheck
+import _isObject from 'lodash/isObject';
+import { createSlice } from '@reduxjs/toolkit/dist/redux-toolkit.esm';
+import fetchReport from './reportActions/fetchReport';
+import Axios from 'axios';
+
+const DEFAULT_TIMEOUT = 60_000;
+
+export default createSlice({
+  name: 'report',
+  initialState: {
+    uid: null,
+    document: null,
+    isFetching: false,
+    isFetchCancelled: false,
```

Comment:
Do we want to support cancelling a fetch? is that necessary?

Reply:
yes, else we end up downloading the same document multiple times